tensorboard/plugins/text/summary.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Text summaries and TensorFlow operations to create them.
+A text summary stores a single string value.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow as tf
+
+from tensorboard.plugins.text import metadata
+
+
+def op(name,
+       data,
+       display_name=None,
+       description=None,
+       collections=None):
+  """Summarizes textual data.
+
+  Text data summarized via this plugin will be visible in the Text Dashboard
+  in TensorBoard. The standard TensorBoard Text Dashboard will render markdown
+  in the strings, and will automatically organize 1D and 2D tensors into tables.
+  If a tensor with more than 2 dimensions is provided, a 2D subarray will be
+  displayed along with a warning message. (Note that this behavior is not
+  intrinsic to the text summary API, but rather to the default TensorBoard text
+  plugin.)
+
+  Args:
+    name: A name for the generated node. Will also serve as a series name in
+      TensorBoard.
+    data: a string-type Tensor to summarize. The text should be UTF-encoded.
```

Comment:
It has to be UTF-8–encoded specifically. UTF-16/UTF-32/UCS-2 will not work. Also, prefer "must" over "should" here for clarity, and TensorFlow convention is to capitalize the initial word (`s/ a/ A `).

Reply:
Done.